cluster-assign-roles.sh

```diff
@@ -172,7 +172,13 @@ function hadoop_install {
   done
 
   printf "Installing heads...\n"
-  install_machines $(printf ${hosts// /\\n} | grep -i "BCPC-Hadoop-Head" | sort)
+  for m in $(printf ${hosts// /\\n} | grep -i "BCPC-Hadoop-Head"| sort); do
+    [[ "$m" =~ $REGEX ]]
+    # authenticate the node one by one
+    vaults=$(sudo ./find_resources.rb ${BASH_REMATCH[3]} | tail -1)
```

Comment:
Since we twice use `${BASH_REMATCH[3]}` could you set this to a descriptive variable? (E.g. nodename?)